packages/augur-simplified/src/modules/market/trading-form.tsx

```diff
@@ -381,90 +418,78 @@ const TradingForm = ({
-      }
-      if (orderType === BUY ) {
-        if (approvals?.trade?.enter[ammCash?.name]) {
-          setIsApprovedTrade(true);
-        } else {
-          setIsApprovedTrade(false);
-        }
-      } else if (orderType === SELL) {
-        if (approvals?.trade?.exit[ammCash?.name]) {
-          setIsApprovedTrade(true);
-        } else {
-          setIsApprovedTrade(false);
-        }
-      }
-    }
-
     const getEstimate = async () => {
-      const outputYesShares = selectedOutcome.id === YES_OUTCOME_ID;
-      if (orderType === BUY) {
-        const breakdown = await estimateEnterTrade(amm, amount, outputYesShares);
-        if (isMounted) setBreakdown(breakdown);
-        if (breakdown && breakdown.outputValue !== "0") {
-          const { slippagePercent, ratePerCash: rate, outputValue } = breakdown;
-          if (isMounted) {
-            const ratePerCash = `1 ${amm?.cash?.name} = ${rate}`;
-            setTradeEstimates({ slippagePercent, ratePerCash, outputAmount: outputValue })
-            updateButtonError('');
-          }
-        }
-      } else {
-        let userBalances = [];
-        const hasShares =
-          balances?.marketShares && balances?.marketShares[amm?.id];
-        if (hasShares) {
-          userBalances = hasShares.outcomeSharesRaw;
-        }
-        const breakdown = await estimateExitTrade(amm, amount, outputYesShares, userBalances);
-        if (isMounted) setBreakdown(breakdown);
-        if (breakdown && breakdown.outputValue !== "0") {
-          const { slippagePercent, ratePerCash: rate, outputValue } = breakdown;
-          if (isMounted) {
-            const ratePerCash = `1 ${amm?.cash?.name} = ${rate}`;
-            setTradeEstimates({ slippagePercent, ratePerCash, outputAmount: outputValue })
-            updateButtonError('');
-          }
+      const outputYesShares = selectedOutcomeId === YES_OUTCOME_ID;
+      let userBalances = [];
+      if (outcomeSharesRaw) {
+        userBalances = marketShares?.outcomeSharesRaw;
+      }
+      const breakdown =
+        orderType === BUY
+          ? await estimateEnterTrade(amm, amount, outputYesShares)
+          : await estimateExitTrade(amm, amount, outputYesShares, userBalances);
+      if (breakdown && breakdown.outputValue !== '0') {
+        const { slippagePercent, ratePerCash: rate, outputValue } = breakdown;
+        if (isMounted) {
+          const ratePerCash = `1 ${amm?.cash?.name} = ${rate}`;
+          setTradeEstimates({
+            slippagePercent,
+            ratePerCash,
+            outputAmount: outputValue,
+          });
         }
       }
+      if (isMounted) setBreakdown(breakdown);
     };
-    if (orderType && selectedOutcome.id && amount && Number(amount) > 0) {
+
+    if (amount && Number(amount) > 0) {
       getEstimate();
-    } else {
-      orderType === BUY ? setBreakdown(undefined) : setBreakdown(undefined);
+    } else if (breakdown !== null) {
+      setBreakdown(null);
```

Comment:
setting Breakdown to null, means insufficient liquidity. for default set to undefined.

Reply:
reverting but we should move away from a pattern that relies on null vs undefined vs some object all meaning different things.